.github/workflows/linux-build-base.yml

```
@@ -24,45 +24,122 @@ on:
+        run: |
+          # Clear ccache statistics (not the cache itself) so final stats reflect only this build
+          # The actual cached compilation results from the warmup build are preserved
+          ccache -z
+
+      - name: Build Velox
+        env:
+          # Tune parallelism based on runner type: self-hosted (16-core) vs GitHub-hosted (4-core)
+          # MAX_HIGH_MEM_JOBS: High-memory compilation tasks (templates, large TUs)
+          # MAX_LINK_JOBS: Parallel linking tasks (memory-intensive)
+          # Self-hosted runners have more cores/RAM, so can handle higher parallelism
+          MAKEFLAGS: >-
```

actionlint (1.7.8)

property "labels" is not defined in object type {arch: string; debug: string; environment: string; name: string; os: string; temp: string; tool_cache: string}

(expression)